src/anonymize_proxy.js

```diff
@@ -15,8 +13,8 @@ const anonymizedProxyUrlToServer = {};
  * anonymous proxy URL or the original URL if it was already anonymous.
  */
 const anonymizeProxy = (proxyUrl, callback) => {
-    const parsedProxyUrl = parseUrl(proxyUrl);
-    if (!parsedProxyUrl.host || !parsedProxyUrl.port) {
+    const parsedProxyUrl = new URL(proxyUrl);
+    if (!parsedProxyUrl.hostname || !parsedProxyUrl.port) {
```

Comment:
this is the same as i noted in the tcp tunnel change, right? port can be empty string and the validation will be triggered